cpp/lib/src/Zotero.cc

```diff
@@ -386,138 +336,222 @@ void MarcFormatHandler::ExtractVolumeYearIssueAndPages(const JSON::ObjectNode &o
-            static RegexMatcher * const doi_matcher(RegexMatcher::RegexMatcherFactory("^DOI:\\s*([0-9a-zA-Z./]+)$"));
-            if (doi_matcher->matched(extra))
-                InsertDOI(&new_record, (*doi_matcher)[1]);
-        } else if (key_and_node.first == "websiteTitle")
-            *website_title = JSON::JSONNode::CastToStringNodeOrDie(key_and_node.first, key_and_node.second)->getValue();
+     const std::string physical_form(node_parameters.physical_form); // Physical Form
+     if (not physical_form.empty()) {
+        if (physical_form == "A")
+            record->insertField("007", "tu");
+        else if (physical_form == "O")
+            record->insertField("007", "cr uuu---uuuuu");
         else
-            LOG_WARNING("unknown key \"" + key_and_node.first + "\" with node type "
-                      + JSON::JSONNode::TypeToString(key_and_node.second->getType()) + "! ("
-                      + key_and_node.second->toString() + "), whole record: " + object_node->toString());
-    }
+            LOG_ERROR("unhandled value of physical form: \"" + physical_form + "\"!");
+     }
+
+     // Authors/Creators
+     const std::string creator_tag((node_parameters.creators.size() == 1) ? "100" : "700");
+     for (const auto creator : node_parameters.creators) {
+          MARC::Subfields subfields;
+          subfields.appendSubfield('a', StringUtil::Join(std::vector<std::string>({creator.last_name, creator.first_name}), ','));
+          subfields.appendSubfield('4', creator.type);
+          subfields.appendSubfield('0', "(DE-576)" + creator.author_ppn);
+          record->insertField(creator_tag, subfields);
+     }
+
+     // Titles
+     const std::string title(node_parameters.title);
+     if (not title.empty())
+         record->insertField("245", { { 'a', title } });
+     const std::string short_title(node_parameters.short_title);
+     if (not short_title.empty())
+         record->insertField("246", { { 'a', short_title } });
+     if (title.empty() and short_title.empty()) {
+         const std::string website_title(node_parameters.website_title);
+         if (not website_title.empty())
+             record->insertField("245", { { 'a', website_title } });
+         else
+             LOG_ERROR("No title found");
+     }
+
+     // Language (inserted uncoditionally)
+     std::string language(node_parameters.language);
+     if (language.empty())
+         language = DEFAULT_LANGUAGE_CODE;
+         record->insertField("041", { {'a', language } });
+
+     // Abstract Note
+     const std::string abstract_note(node_parameters.abstract_note);
+     if (not abstract_note.empty())
+         record->insertField("520", { {'a', abstract_note } }, '3' /* indicator 1*/);
+
+     // FIXME Which date is this??
+     const std::string date(node_parameters.date);
+     if (not date.empty())
+         record->insertField("362", { {'a', date} });
```

Comment:
where is date normalization performed?
JSON => struct?
struct => MARC?
MARC => Augmented MARC?
(haven't found it anywhere)